### PR TITLE
Pfp

### PR DESCRIPTION
**Pull request by mlooney**

### CODE REVIEW

File: spec/lib/tenejo/preflight_spec.rb [hunk 1]
```diff
@@ -142,6 +142,22 @@
     end
   end
   describe Tenejo::PFWork do
+    context "a record with a potentially packed field, which doesn't have a packed value" do
+      let(:rec) { described_class.new({:resource_type=>"Book"}, 1) }
+      it "should record warning " do
```
Comment (mark-dce) [inline spec/lib/tenejo/preflight_spec.rb:147]: Is this a copy-paste error.  Doesn't look like this give a warning - and it shouldn't.

`it "should wrap the value in an array" do`?

File: spec/lib/tenejo/preflight_spec.rb [hunk 2]
```diff
@@ -142,6 +142,22 @@
...
+        expect(rec.valid?).to eq false
```
Comment (mark-dce) [inline spec/lib/tenejo/preflight_spec.rb:148]: This looks valid to me.  I don't understand this expectation.

File: app/lib/tenejo/pf_object.rb [hunk 3]
```diff
@@ -14,8 +14,9 @@ def initialize(h, lineno)
       self.lineno = lineno
       h.delete(:object_type)
       h.each do |k, v|
-        if respond_to?("#{k}=")
-          send("#{k}=", v) if v.present?
+        if respond_to?("#{k}=") && v.present?
+          v = v.split("|~|") if v =~ /\|~\|/
```
Comment (mark-dce) [inline app/lib/tenejo/pf_object.rb:18]: what about 
```
v = v.split("|~|")
if should_be_singular?(k)
    rec.warnings[k] << "#{k} on line #{rec.lineno} has extra values that will be ignored" if v.count > 1
    v = v.first
end
```

File: spec/lib/tenejo/preflight_spec.rb (hunk 2 above)
Comment (mlooney) [reply spec/lib/tenejo/preflight_spec.rb:148]: I call valid? to trigger validation. the record is invalid for reasons not covered in this particular test case

**Pull request closed without merging**